operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java

```diff
@@ -103,16 +109,31 @@ public UpdateControl<Keycloak> reconcile(Keycloak kc, Context<Keycloak> context)
 
         Log.infof("--- Reconciling Keycloak: %s in namespace: %s", kcName, namespace);
 
+        boolean modifiedSpec = false;
         if (kc.getSpec().getInstances() == null) {
             // explicitly set defaults - and let another reconciliation happen
             // this avoids ensuring unintentional modifications have not been made to the cr
             kc.getSpec().setInstances(1);
+            modifiedSpec = true;
+        }
+        if (kc.getSpec().getIngressSpec() != null && kc.getSpec().getIngressSpec().isIngressEnabled()
```

Comment:
After the recent changes, we now require ingress class name to be always set. [Before](https://github.com/keycloak/keycloak/blob/9d0b884cef189f5440bf22805cd099bce7ce6ea0/operator/src/main/java/org/keycloak/operator/controllers/KeycloakController.java#L116) you could omit the ingress spec completely. I wonder if it might be quite user friendly if we detect the default IngressClass on the cluster and if it matches `openshift-default`, we would not require users to explicitly specify it. WDYT @shawkins?

Reply:
The ingress class name isn't considered required (yet).

You could consider setting it automatically, that will work most of the time.  It will be confusing though if/when it changes and then there's behavioral differences due to implementation details in the ingress.  We ran into a similar case with the notion of the default storage class with managed kafka - assuming the default seemed fine, until the default changed across openshift versions.

Reply:
Yeah, I would not assume what's the default IngressClass. What I meant is that maybe we could actively check it in the Operator.

Reply:
Sorry I'm not describing the situation fully.  One proposal for specifying the storage class in managed kafka was just to use the system default if it wasn't specified in the cr - mostly because the assumption was that the system default matched what had been tested / approved for use and there would be less to capture about cross-platform support. However openshift 4.something changed their default storage class from from gp2 to gp3, which was not something that had been tested / approved, so the system default was considered too much of a free variable from a management perspective.

As long as there's documentation somewhere saying that we're automatically filling in the ingress classname based upon our initial observation at the time of deployment such that an update that omits the classname, or subsequent deployments may pick up a different default.